src/ProjectTempoListener.cpp

```diff
@@ -0,0 +1,27 @@
+#include "ProjectTempoListener.h"
+#include "Project.h"
+#include "ProjectTimeSignature.h"
+#include "WaveTrack.h"
+
+ProjectTempoListener::ProjectTempoListener(
```

Comment:
I think the listener and publisher classes should just be fused into one.

Reply:
But that'd mean moving the implementation of the publisher down in `src`, meaning more code to re-implement when migrating to Qt.